### PR TITLE
Optimize pattern generation

### PR DESCRIPTION
**Pull request by masatake**
Making a pattern takes long time for long input line.  This commit
shortens the time. See the second commit about evaluation of my technique.

### CODE REVIEW

File: main/entry.c [hunk 1]
```diff
@@ -88,7 +88,8 @@ tagFile TagFile = {
 	    .queue = NULL,
 	    .length = 0,
 	    .count  = 0
-    }
+    },
+    .patternCacheValid = FALSE,
```
Comment (vhda) [inline main/entry.c]: Strange indentation

Comment (vhda) [inline main/entry.c]: Ignore me! Everything is fine! :)

**Comment by masatake** [comment]
Commit log messages are modified.

**Comment by masatake** [comment]
Now, ctags can handle very long single line JavaScript file.
However, there is not test case for such file. #140 must be solved.

**Pull request merged**